__tests__/UsingTypeScript.test.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { StrictMode, ChangeEvent, forwardRef } from 'react';
+import { view as v, store as s } from 'react-easy-state';
+import { render, cleanup } from '@testing-library/react/pure';
+
+const store = s({
+  name: 'Peter',
+  setNameEvent(ev: ChangeEvent<HTMLInputElement>) {
+    if (ev.target) {
+      this.name = ev.target.value;
+    }
+  },
+});
+
+// properly retain these props for the views
+type AppProps = {
+  greeting: string;
+};
+
+const App = v(
+  forwardRef<HTMLLabelElement, AppProps>(
+    ({ greeting, children }, ref) => (
+      <div>
+        <h1>
+          {greeting} {store.name}!
+        </h1>
+        <label ref={ref} htmlFor="name">
+          Name:
+        </label>
+        <input
+          id="name"
+          value={store.name}
+          onChange={store.setNameEvent.bind(store)}
+        />
+        {children}
+      </div>
+    ),
+  ),
+);
+
+describe('Using TypeScript App', () => {
+  afterAll(() => {
+    cleanup();
+  });
+
+  test('should render without warnings', () => {
+    const { container } = render(
+      <StrictMode>
+        <App greeting="Hello" />
```

Comment:
You do not pass a ref here. I don't know if it affects the validity of the forwardRef test (probably not, just curious).